src/runtime/pkg/katautils/config.go

```diff
@@ -244,11 +245,25 @@ func (h hypervisor) jailerPath() (string, error) {
 	return ResolvePath(p)
 }
 
+func (h hypervisor) igvm() (string, error) {
+	p := h.Igvm
+
+	if p == "" {
+		return "", nil
+	}
+
+	return ResolvePath(p)
+}
+
 func (h hypervisor) kernel() (string, error) {
 	p := h.Kernel
 
 	if p == "" {
-		p = defaultKernelPath
+		if h.Igvm == "" {
```

Comment:
Minor: Is there a chance to leave this section unchanged with IGVM? When using IGVM, potentially calling into kernel() should not be done at all or should have no effect.

Reply:
Even if no kernel is set in the config, this path will be taken. In that case, a default path (defaultKernelPath) is set instead of having it be a Null value, and if it's set it will be checked later on in the code